### ISSUE TITLE
Submitting LSF job for the mirtrace process in test causes Bad Job name error

### ISSUE THREAD

**Issue by Accio** [nf-core/smrnaseq#109]
## Check Documentation

I have checked the following places for your error:

- [x] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [x] [nf-core/smrnaseq pipeline documentation](https://nf-co.re/nf-core/smrnaseq/usage)

## Description of the bug

When starting the pipeline with `nextflow run nf-core/smrnaseq -profile test,singularity`, local runs causes no problem. However, if we set the executor to be LSF, it reports an error, caused by bsub, with the output `Bad job name. Job not submitted.`. We believe the problem is caused by the mirtrace process, which prints `mirtrace ([sample_1.fastq.gz, sample_2.fastq.gz, sample_3.fastq.gz])`.

## Steps to reproduce

Steps to reproduce the behaviour:

1. Set "lsf" as process.executor in config.
2. Command line: `nextflow run nf-core/smrnaseq -profile test,singularity`
2. See error: 
```
Error executing process > 'mirtrace ([sample_1.fastq.gz, sample_2.fastq.gz, sample_3.fastq.gz])'

Caused by:
  Failed to submit process to grid scheduler for execution

Command executed:

  bsub

Command exit status:
  255

Command output:
  Bad job name. Job not submitted.

```

We believe that the following job name, which contains special characters `[` and `]`, may cause the problem with LSF:

`
#BSUB -J "nf-mirtrace_([sample_1.fastq.gz,_sample_2.fastq.gz,_sample_3.fastq.gz])"
`
## Expected behaviour

It should run error-free as local runs.

## Log files

Have you provided the following extra information/files:

- [ ] The command used to run the pipeline
- [ ] The `.nextflow.log` file 

## System

- Hardware:  HPC 
- Executor: LSF
- OS: CentOS Linux...]
- Version 

## Nextflow Installation

- Version: 20.10 

## Container engine

- Engine: Singularity
- version: 
- Image tag:

**Comment by lpantano** [comment]
we will try to fix that in the next version #104

**Comment by apeltzer** [comment]
Hi @Accio ! 

Can you give `dev` another try? `-r dev` when running the pipeline, ideally with a newer Nextflow version (> 21.04.2 should work fine).
